### ISSUE TITLE
Feature annotation extension

### ISSUE THREAD

**Issue by stanstrup** [sneumann/xcms#295]
Is there a way to annotate features like you have pData for the files?
This could allow closer integration with CAMERA or simple mz/rt annotation, but also allow custom functions to for example flag possible contaminants.

I guess it is not possible right now but could it be a good idea?

**Comment by jotsetung** [comment]
good point. problem is that the `MSnExp`/`OnDiskMSnExp`/`XCMSnExp` objects have already a `featureData` slot (where features are however spectra, not features in our meaning...). But we could add a `featureAnnotation` slot.
